public/app/core/utils/dag.test.ts

```diff
@@ -90,6 +97,12 @@ describe('Directed acyclic graph', () => {
       expect(actual).toEqual(expect.arrayContaining([nodeF, nodeG]));
     });
 
+    it('when optimizing input edges for node 0 should return node A', () => {
+      const actual = node0.getOptimizedInputEdges().map((e) => e.inputNode);
+      expect(actual).toHaveLength(1);
+      expect(actual).toEqual(expect.arrayContaining([nodeA]));
+    });
```

Comment:
The previous implementation of `getOptimizedInputEdges` failed this test case b/c the transitive reduction on the graph only went to a finite depth instead of traversing the entire graph to check for alternative paths back to the source node.